src/Pages/GlobalConfigurations/Authorization/shared/components/AppPermissions/AppPermissions.component.tsx

```diff
@@ -0,0 +1,903 @@
+    const { serverMode } = useContext(mainContext)
+    const {
+        directPermission,
+        setDirectPermission,
+        setChartPermission,
+        setK8sPermission,
+        currentK8sPermissionRef,
+        data,
+    } = usePermissionConfiguration()
+    const { customRoles } = useAuthorizationContext()
+    const { isSuperAdmin: superAdmin } = useMainContext()
+    const { url, path } = useRouteMatch()
+    const location = useLocation()
+
+    const [isLoading, setIsLoading] = useState(false)
+    const [appsList, setAppsList] = useState<AppPermissionsDetailType['appsList']>(new Map())
+    const [appsListHelmApps, setAppsListHelmApps] = useState<AppPermissionsDetailType['appsListHelmApps']>(new Map())
+    const [jobsList, setJobsList] = useState<AppPermissionsDetailType['jobsList']>(new Map())
+
+    const [isDataLoading, configData, error, reload] = useAsync(() =>
+        Promise.all([
+            getProjectList(),
+            getEnvironmentListMin(),
+            serverMode === SERVER_MODE.EA_ONLY ? null : getChartGroups(),
+            getEnvironmentListHelmApps(),
+        ]),
+    )
+
+    const isNonEAMode = serverMode !== SERVER_MODE.EA_ONLY
+    const projectsList = configData?.[0]?.result ?? []
+    const environmentsList = configData?.[1]?.result ?? []
+    const chartGroupsList = configData?.[2]?.result?.groups ?? []
+    const envClustersList = configData?.[3]?.result ?? []
+
+    const environmentClusterOptions = getEnvironmentClusterOptions(envClustersList)
+
+    const _getEnvironmentOptions = (entity: DirectPermissionRow['permission']['entity']) =>
+        getEnvironmentOptions(environmentsList, entity)
+
+    const appPermissionDetailConfig = getAppPermissionDetailConfig(path, serverMode)
+    const navLinksConfig = getNavLinksConfig(serverMode, superAdmin)
+
+    async function fetchJobsList(projectIds: number[]) {
+        const missingProjects = projectIds.filter((projectId) => !jobsList.has(projectId))
+        if (missingProjects.length === 0) {
+            return
+        }
+        setJobsList((_jobsList) =>
+            missingProjects.reduce((__jobsList, projectId) => {
+                __jobsList.set(projectId, { loading: true, result: [], error: null })
+                return __jobsList
+            }, _jobsList),
+        )
+        try {
+            // TODO (v3): Temporary workaround since projectId are not available in job list
+            missingProjects.forEach(async (projectId) => {
+                const {
+                    result: { jobContainers },
+                } = await getJobs({ teams: [projectId] })
+
+                setJobsList((_jobsList) => {
+                    _jobsList.set(+projectId, {
+                        loading: false,
+                        result: jobContainers || [],
+                        error: null,
+                    })
+                    return _jobsList
+                })
+            })
+            // const {
```

Comment:
Plesae remove commented code

Reply:
Have removed it in subsequent PR: https://github.com/devtron-labs/dashboard/pull/1650/files#diff-d04fd01e3bfe62a8dec22cd053484d718dac185eab0e987f5e0062a800ca5bc9R131